### PR TITLE
Add document

### PR DESCRIPTION
**Pull request by qwe661234**

### CODE REVIEW

File: Document.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+### Command
+* Server
+> $ sudo perf trace ./downloads/lighttpd1.4-lighttpd-1.4.58/src/lighttpd -D -f downloads/lighttpd1.4-lighttpd-1.4.58/src/lighttpd.conf
+* Client
+> $ ab -n 10 -c 10 -k http://127.0.0.1:3000/
+
+## Reference 
+* [GitHub: ESCA](https://github.com/eecheng87/ESCA)
+* [ESCA thesis](https://eecheng87.github.io/ESCA/main.pdf)
+* [System Call Note](https://hackmd.io/@combo-tw/Linux-%E8%AE%80%E6%9B%B8%E6%9C%83/%2F%40a29654068%2FHyD4Lu_Dr)
```
Comment (jserv) [inline Document.md:173]: Don't reference to HackMD notes because they might be changed frequently and unexpectedly.

File: Document.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
+# Effective System Call Aggregation(ESCA) Document
+## ESCA Design
+The code section enclosed by `batch_start()` and `batch_flush()` is called batching segment. It can appear more than one time in a single application. Compared with typical syscalls, ESCA eliminates mode switches in batching segments by decoupling syscalls. Instead of switching to the kernel or executing the corresponding service routine, syscalls in batching segment only record their syscall ID and arguments in the shared table.After batch_flush is called, ESCA finally switches to kernel mode, executes all syscalls in the shared table, and then switches back to user mode.
+## Typical system call flow
+### Overview
+![](https://i.imgur.com/zEXqLKH.png)
```
Comment (jserv) [inline Document.md:6]: Put the diagrams in this repository.

File: Document.md (hunk 1 above)
Comment (qwe661234) [reply Document.md:173]: Ok, I removed it.

File: Document.md (hunk 2 above)
Comment (qwe661234) [reply Document.md:6]: done

**Pull request merged**